allauth/socialaccount/providers/amazon/provider.py

```diff
@@ -5,7 +5,7 @@
 
 class AmazonAccount(ProviderAccount):
     def to_str(self):
-        return self.account.extra_data.get("name", super(AmazonAccount, self).to_str())
+        return self.account.extra_data.get("email", super().to_str())
```

Comment:
As above.

Reply:
If `email` is falsey, then `super().to_str()` will probably grab the `name` value.

Reply:
But here as well, `email` is already being returned by the super, so no need to provide a `to_str` anymore.